### ISSUE TITLE
Android build error: JS sourcemap file doesn't exist, aborting upload

### ISSUE THREAD

**Issue by bennettp123** [DataDog/dd-sdk-reactnative#528]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `@datadog/mobile-react-native@1.8.4` for the project I'm working on.

**Describe what happened**

When trying to upload sourcemaps, it failed with the following error:

```
Script '${PROJECT}/node_modules/@datadog/mobile-react-native/datadog-sourcemaps.gradle' line: 87

JS sourcemap file doesn't exist, aborting upload
```

**Steps to reproduce the issue:**

`cd ./android && ./gradlew bundleDevRelease --no-daemon`

**Describe what you expected:**

I expected it to upload the sourcemaps at `/Users/bennett/src/west-app/android/app/build/generated/sourcemaps/react/dev/release/index.android.bundle.map`

**Additional context**

-   Version of the SDK: `@datadog/mobile-react-native@1.8.4`
-   Version of React Native: `react-native@0.71.12`
-   [`package.json`](https://gist.github.com/bennettp123/e57cae5765e389aff5795aa38700ac30#file-package-json) and [`yarn.lock`](https://gist.github.com/bennettp123/e57cae5765e389aff5795aa38700ac30#file-yarn-lock)
-   [`Podfile`](https://gist.github.com/bennettp123/e57cae5765e389aff5795aa38700ac30#file-ios-podfile) and [`Podfile.lock`](https://gist.github.com/bennettp123/e57cae5765e389aff5795aa38700ac30#file-ios-podfile-lock)
-   [`android/build.gradle`](https://gist.github.com/bennettp123/e57cae5765e389aff5795aa38700ac30#file-android-build-gradle) and [`android/app/build.gradle`](https://gist.github.com/bennettp123/e57cae5765e389aff5795aa38700ac30#file-android-app-build-gradle)

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/@datadog/mobile-react-native/datadog-sourcemaps.gradle b/node_modules/@datadog/mobile-react-native/datadog-sourcemaps.gradle
index 6c53800..70ebef1 100644
--- a/node_modules/@datadog/mobile-react-native/datadog-sourcemaps.gradle
+++ b/node_modules/@datadog/mobile-react-native/datadog-sourcemaps.gradle
@@ -28,7 +28,7 @@ afterEvaluate {
 
         // locations and names are defined in "../../node_modules/react-native/react.gradle"
         def targetName = variant.name.capitalize() // e.g. Release
-        def targetPath = variant.dirName // e.g. release
+        def targetPath = variant.name // e.g. devRelease
 
         def reactConfig = getReactConfig(buildDir, targetName, targetPath)
         def reactRoot = file(reactConfig.root)
```

**Comment by louiszawadzki** [comment]
Hi @bennettp123, thanks for reaching out!

Thanks for providing all this input, I've been able to reproduce the issue on my end and I am working on a fix.
I'll let you know once it is released.

**Comment by louiszawadzki** [comment]
Hi @bennettp123, 

We released version `1.8.5` of the SDK that contains a fix for your problem. 
I'll close the issue for now, feel free to reopen it if the issue isn't fixed!
